File: src/net/dinkla/raytracer/btdf/FresnelTransmitter.java
package net.dinkla.raytracer.btdf;

import net.dinkla.raytracer.colors.Color;
import net.dinkla.raytracer.hits.Shade;
import net.dinkla.raytracer.math.Vector3DF;

/**
 * Created by IntelliJ IDEA.
 * User: Jörn Dinkla
 * Date: 15.05.2010
 * Time: 11:19:51
 * To change this template use File | Settings | File Templates.
 */
public class FresnelTransmitter<C extends Color> extends BTDF<C> {
    
    @Override
    public C f(Shade sr, Vector3DF wo, Vector3DF wi) {
        return null;
    }

    @Override
    public boolean isTir(Shade sr) {
        return false;
    }

    @Override
    public C rho(Shade sr, Vector3DF wo) {
        return null;
    }

    @Override
    public Sample sampleF(Shade sr, Vector3DF wo) {
        return null;
    }
    
}

File: src/jvmMain/kotlin/net/dinkla/raytracer/btdf/FresnelTransmitter.kt
package net.dinkla.raytracer.btdf

import net.dinkla.raytracer.btdf.BTDF.Sample
import net.dinkla.raytracer.colors.Color
import net.dinkla.raytracer.hits.Shade
import net.dinkla.raytracer.math.Vector3D

class FresnelTransmitter : BTDF {

    override fun f(sr: Shade, wo: Vector3D, wi: Vector3D): Color = Color.BLACK

    override fun isTir(sr: Shade): Boolean = false

    override fun rho(sr: Shade, wo: Vector3D): Color = Color.BLACK

    override fun sampleF(sr: Shade, wo: Vector3D): Sample = Sample()

}
